subsys/usb/usb_device.c

```diff
@@ -1542,18 +1544,24 @@ int usb_enable(void)
 		return 0;
 	}
 
+	/* Prevent from calling usb_enable form different contex.
+	 * This should only be called once.
+	 */
+	LOG_DBG("lock usb_enable_lock mutex");
+	k_mutex_lock(&usb_enable_lock, K_FOREVER);
```

Comment:
The above check for `usb_dev.enabled` must be moved after the mutex is acquired, or two threads could pass that state, then sequentially both initialize the device.

Reply:
Good point. I missed this case.

Reply:
Guess it was my fault too.